test/test_nn.py

```diff
@@ -11486,7 +11408,8 @@ def run_test_case(norm_type, error_if_nonfinite, scalar, grad_only_one_elem, pre
 
     @onlyCUDA
     @deviceCountAtLeast(2)
-    def test_clip_grad_norm_multi_device(self, devices):
+    @parametrize_test('foreach', (False, True))
+    def test_clip_grad_norm_multi_device(self, devices, foreach):
```

Comment:
Not a concern with your PR, but I am realizing we never run this in CI because we only have one CI config where there is more than one GPU and we don't run this test in that config. 🤔 Filed https://github.com/pytorch/pytorch/issues/92173

Reply:
If you want to see this run in CI, you would have to add a line similar to https://github.com/pytorch/pytorch/pull/92048/files#diff-5356a2d45f3d28e01b954926d7f1681cccb0c9ad2cafd722478de8c283090bd3R48

Reply:
And then you would also need to add the ciflow/periodic label to get the multigpu tests to trigger.

Reply:
ok my bad they actually run as expected with `pytest -k clip_grad test/test_nn.py`, not sure what I did wrong before